src/utils/__tests__/makeBackoffMachine-test.js

```diff
@@ -24,17 +24,28 @@ describe('makeBackoffMachine', () => {
     return Date.now() - start;
   };
 
-  test('timeouts are 100ms, 200ms, 400ms, 800ms...', async () => {
-    const expectedDurations = [100, 200, 400, 800, 1600, 3200, 6400, 10000, 10000, 10000, 10000];
-    const results: number[] = [];
-
-    const backoffMachine = makeBackoffMachine();
-    for (let j = 0; j < expectedDurations.length; j++) {
-      const duration = await measureWait(backoffMachine.wait());
-      results.push(duration);
+  test('timeouts are random from zero to 100ms, 200ms, 400ms, 800ms...', async () => {
+    const NUM_TRIALS = 100;
+    const expectedMaxDurations = [100, 200, 400, 800, 1600, 3200, 6400, 10000, 10000, 10000, 10000];
+
+    const trialResults: Array<number[]> = [];
+
+    for (let i = 0; i < NUM_TRIALS; i++) {
+      const resultsForThisTrial = [];
+      const backoffMachine = makeBackoffMachine();
+      for (let j = 0; j < expectedMaxDurations.length; j++) {
+        const duration = await measureWait(backoffMachine.wait());
+        resultsForThisTrial.push(duration);
+      }
+      trialResults.push(resultsForThisTrial);
     }
-    expectedDurations.forEach((expectedDuration, i) => {
-      expect(results[i]).toBe(expectedDuration);
+
+    expectedMaxDurations.forEach((expectedMax, i) => {
+      const maxFromAllTrials = Math.max(...trialResults.map(r => r[i]));
+      const minFromAllTrials = Math.min(...trialResults.map(r => r[i]));
+
+      expect(minFromAllTrials).toBeLessThan(expectedMax * 0.25);
+      expect(maxFromAllTrials).toBeGreaterThan(expectedMax * 0.75);
```

Comment:
> Which is of course wrong. But perhaps this could use a comment noting that the chances of this flaking are on the order of `2 * (1-(1-(0.25**100))**11)` ≈ `1 / 7.3e58`?

I have no idea what I was thinking here with `0.25**100` rather than `0.75**100`. (The rest is due to considering the aggregate flake chances rather than individual assertion chance failure. I'm skeptical of `2 *`, in retrospect, but it's probably at least close to that.)

... anyway, the comment you've added in the current head is :+1:.

Reply:
> ... anyway, the comment you've added in the current head is 👍.

Many thanks to @gnprice for this, I can't take the credit! 😄